WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -897,8 +899,15 @@ final class EditableOrderViewModel: ObservableObject {
     }
 
     func onAddCustomAmountButtonTapped() {
-        editingFee = nil
         analytics.track(.orderCreationAddCustomAmountTapped)
+        addCustomAmount()
+    }
+
+    /// Starts the flow to add a custom amount.
+    func addCustomAmount() {
+        editingFee = nil
+        enableAddingCustomAmountViaOrderTotalPercentage ?
+        customAmountsSectionViewModel.showAddCustomAmountOptionsDialog.toggle() : customAmountsSectionViewModel.showAddCustomAmount.toggle()
```

Comment:
Does it feel odd to keep both `onAddCustomAmountButtonTapped` and `addCustomAmount` access control internal just for the sake of the track event? I don't see a reason why'd want to tap without tracking. WDYT if we move the analytics call to a private method instead?

```suggestion
    func addCustomAmount() {
        editingFee = nil
        enableAddingCustomAmountViaOrderTotalPercentage ?
        customAmountsSectionViewModel.showAddCustomAmountOptionsDialog.toggle() : customAmountsSectionViewModel.showAddCustomAmount.toggle()
        
        trackOrderCreationAddCustomAmountTapped() 
}

private func trackOrderCreationAddCustomAmountTapped() {
        analytics.track(.orderCreationAddCustomAmountTapped)
        }        
```

With this we can also edit `test_onAddCustomAmountButtonTapped_then_it_tracks_event`